chapters/05-comparison.Rmd

```diff
@@ -0,0 +1,195 @@
+
+For all implementations that use binning $2^10 = 1024$ bins were used. This is the default used in KDEpy, a power of $2$ (which is favorable for FFT based algorithms), results in an exact kernel density calculation for the lowest sample size used ($10^3$) but also yields results with high accuracy for the highest sample size used ($10^8$). 
+
+```{python showDistributions, echo=FALSE, eval=TRUE, fig.cap="Distributions used for the comparisons", fig.align='center', out.width='80%'}
+figure, axes = runner.plot_distributions(distributions_to_evaluate, xlim)
+figure
+```
+
+## Differences of Exact, Binned, FFT and ISJ implementations 
+
+First, the exact kernel density estimation implementation is compared against linearly binned, FFT and ISJ implementations run on a Macbook Pro 2013 Retina using the CPU.
+
+The sample sizes lie in the range of $10^3$ to $10^4$. The number of samples is restricted because calculating the exact kernel density estimation for more than $10^4$ kernels is computationally unfeasible (larger datasets would lead to an exponentially larger runtime).
+
+### Accuracy
+
+Plotted below are the comparisons for sample size of $10^4$.
+
+```{python compareSimpleBinnedFFTISJEstimations, echo=FALSE, eval=TRUE, fig.cap="Comparison between the four algorithms 'Exact', 'Binned', 'FFT' and 'ISJ' with $n=10^4$ sample points", fig.align='center', out.width='80%'}
+figure, axes = runner.plot_estimations(restricted_estimations, distributions_to_evaluate, 1e4, ['ZfitExact', 'ZfitBinned', 'ZfitFFT', 'ZfitISJ'])
+figure
+```
+
+It becomes obvious that the ISJ approach is especially favorable for complicated spiky distributions like the two bottom ones. We can see this in more detail below. Using the ISJ the integrated square error (ISE) is an order of magnitude lower.
+
+```{python compareSimpleBinnedFFTISJEstimationClaw, echo=FALSE, eval=TRUE, fig.cap="Comparison between the four basic algorithms 'Exact', 'Binned', 'FFT', 'ISJ' with $n=10^4$ sample points on distribution 'Claw'", fig.align='center', out.width='80%'}
+figure, axes = runner.plot_estimation(restricted_estimations, 'Claw', ['ZfitExact', 'ZfitBinned', 'ZfitFFT', 'ZfitISJ'], 1e4)
+```
+
+The calculated integrated square errors for all distributions are as follows:
+
+```{python compareSimpleBinnedFFTISJErrors, echo=FALSE, eval=TRUE, fig.cap="Integrated square errors (ISE) for the four algorithms 'Exact', 'Binned', 'FFT' and 'ISJ' with $n=10^4$ sample points", fig.align='center', out.width='80%'}
+figure, axes = runner.plot_integrated_square_errors(restricted_estimations, distributions_to_evaluate, 1e4, ['ZfitExact', 'ZfitBinned', 'ZfitFFT', 'ZfitISJ'])
+figure
+```
```

Comment:
[10] comment on this! What do we see in the plots, why are they this way? E.g. why is the ISJ so "bad" for the gaussian case (or rather, the other ones so good?), where does it shine. The plot is just the information, add the explanation. You're the expert, tell us something about what we see here, make conclusions from it